modules/core/shared/src/main/scala/lucuma/core/math/Constants.scala

```diff
@@ -4,27 +4,49 @@
 package lucuma.core.math
 
 import java.time.Duration
+import coulomb._
+import coulomb.physicalconstants._
+import lucuma.core.math.units.MetersPerSecond
+import spire.math.Rational
+import coulomb.define.DerivedUnit
+import coulomb.si.Meter
+import coulomb.physicalconstants.infra.PhysicalConstantQuantity
 
 object Constants {
+  object infra {
+    trait AstronomicalUnit
+    implicit val defineAU = {
+      val coef = Rational(149597870660L)
+      DerivedUnit[AstronomicalUnit, Meter](abbv = "AU", coef = coef)
+    }
+
+    trait EquatorialRadius
+    implicit val defineEquatorialRadius = {
+      val coef = Rational(6378137L)
+      DerivedUnit[EquatorialRadius, Meter](abbv = "equatorial-radius", coef = coef)
+    }
+  }
+  import infra._
 
   /*
    * Astronomical Constants
    */
 
   /** One AU in meters. */
-  val AstronomicalUnit: Long = 149597870660L
+  def AstronomicalUnit[V](implicit
+    pcq: PhysicalConstantQuantity[V, AstronomicalUnit]
+  ): Quantity[V, pcq.QU] = pcq.q
 
   /** Meters/sec in 1 AU/day. */
   val MetersPerSecondInAUPerDay: Double = 1731456.83633
 
   /** Flattening of earth, 1/298.257 */
   val FlatteningOfEarth: Double = 0.003352813
 
-  /** Equatorial radius of Earth in meters. */
-  val EquatorialRadiusMeters: Int = 6378137
-
-  /** 1 AU in meters. */
-  val MetersInAU: Double = 1.4959787066e11
+  /** Equatorial radius of Earth. */
+  def EquatorialRadius[V](implicit
+    pcq: PhysicalConstantQuantity[V, EquatorialRadius]
+  ): Quantity[V, pcq.QU] = pcq.q
```

Comment:
Why can't this be `Quantity[Int, Meters]` ?

Reply:
This is the idiomatic encoding for constants used in coulomb, the idea is you can select the type most appropriate for your particular caller but the units are defined in the constant